apps/dashboard/features/token-distribution/components/TokenDistributionMetrics.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from "react";
+import { CardTitle } from "@/shared/components/ui/card";
+import { X } from "lucide-react";
+import { TokenDistributionDialog } from "@/features/token-distribution/components";
+import { Button } from "@/shared/components/ui/button";
+import { MetricTypesEnum } from "@/shared/types/enums/metric-type";
+import {
+  DaoMetricsDayBucket,
+  Metric,
+  MetricWithKey,
+} from "@/shared/dao-config/types";
+import { formatChartVariation } from "@/features/token-distribution/utils";
+import { formatNumberUserReadable, formatVariation } from "@/shared/utils";
+
+interface TokenDistributionMetricsProps {
+  appliedMetrics: MetricTypesEnum[];
+  setAppliedMetrics: React.Dispatch<React.SetStateAction<MetricTypesEnum[]>>;
+  setHoveredMetricKey: React.Dispatch<
+    React.SetStateAction<MetricTypesEnum | null>
+  >;
+  timeSeriesData?: Record<MetricTypesEnum, DaoMetricsDayBucket[]> | null;
+  metricsSchema: Record<MetricTypesEnum, Metric>;
+}
+
+const formatMetricsByCategory = (
+  metrics: Record<MetricTypesEnum, Metric>,
+): Record<string, MetricWithKey[]> => {
+  return Object.entries(metrics).reduce(
+    (grouped, [key, metric]) => {
+      const metricKey = key as MetricTypesEnum;
+      const { category } = metric;
+
+      if (!grouped[category]) {
+        grouped[category] = [];
+      }
+
+      grouped[category].push({ ...metric, key: metricKey });
+
+      return grouped;
+    },
+    {} as Record<string, MetricWithKey[]>,
+  );
+};
+
+export const TokenDistributionMetrics = ({
+  appliedMetrics,
+  setAppliedMetrics,
+  metricsSchema,
+  setHoveredMetricKey,
+  timeSeriesData,
+}: TokenDistributionMetricsProps) => {
```

Comment:
The formatChartVariation function is called with timeSeriesData! using non-null assertion, but timeSeriesData can be null according to the type definition. This will cause a runtime error when timeSeriesData is null.